### PR TITLE
ERT Configuration

### PR DESCRIPTION
**Pull request by Kyep**
🆑 Kyep
add: ERTs now support janitor, paranormal, and cyborg ERT units, as well as specific slot configurations.
/🆑

New features:
- ERT dispatch is now handled via a panel, rather than a series of prompts.
- The dispatch window lets you set how many slots there should be of each class. So you can send an all-sec or all-medical or all-eng ERT if you want. Or mix up the types. You can also give people a mixture of type options.
- The Janitor, Paranormal and Cyborg ERT types are now supported. All are disabled (no slots) by default. Sending Cyborgs in a Gamma ERT makes them into Gamma ERT borgs.
- ERT no longer permits signups from job banned people.

ert_panel
ert_notification
ert_choices
ert_janitor
ert_paranormal
ert_borg
ert_borg_gamma

### CODE REVIEW

**Comment by Necaladun** [comment]
Throw a screenshot up of that panel 'cause dear god is it sexy

**Comment by Kyep** [comment]
@Necaladun tons of screenshots added.

File: code/modules/nano/modules/ert_manager.dm [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+	name = "ERT Manager"
+	var ert_type = "Code Red"
+	var commander_slots = 1
+	var security_slots = 3
+	var medical_slots = 3
+	var engineering_slots = 3
+	var janitor_slots = 0
+	var paranormal_slots = 0
+	var cyborg_slots = 0
+	var autoclose = 0
```
Comment (Birdtalon) [inline code/modules/nano/modules/ert_manager.dm]: Spaces?

**Comment by Necaladun** [comment]
Oh and to be clear, we can still just do a lazy whatever ERT?

File: code/game/response_team.dm [hunk 2]
```diff
@@ -141,17 +107,37 @@ var/ert_request_answered = 0
...
 		class = input(src, "Which loadout would you like to choose?") in active_team.get_slot_list()
 		if(!active_team.check_slot_available(class)) // Because the prompt does not update automatically when a slot gets filled.
 			class = 0
 
+	if(class == "Cyborg")
+		active_team.reduceCyborgSlots()
+		var cyborg_type = active_team.getCyborgType()
+		var/mob/living/silicon/robot/ert/R = new cyborg_type()
+		R.forceMove(spawn_location)
+		var rnum = rand(1,1000)
```
Comment (Birdtalon) [inline code/game/response_team.dm]: Space?

File: code/game/response_team.dm [hunk 3]
```diff
@@ -141,17 +107,37 @@ var/ert_request_answered = 0
...
 
 	var/class = 0
 	while(!class)
 		class = input(src, "Which loadout would you like to choose?") in active_team.get_slot_list()
 		if(!active_team.check_slot_available(class)) // Because the prompt does not update automatically when a slot gets filled.
 			class = 0
 
+	if(class == "Cyborg")
+		active_team.reduceCyborgSlots()
+		var cyborg_type = active_team.getCyborgType()
```
Comment (Birdtalon) [inline code/game/response_team.dm]: Space?

File: code/game/response_team.dm [hunk 4]
```diff
@@ -112,6 +76,8 @@ var/ert_request_answered = 0
 
 	// Respawnable players get first dibs
 	for(var/mob/dead/observer/M in ert_candidates)
+		if(jobban_isbanned(M, ROLE_TRAITOR) || jobban_isbanned(M, "Security Officer") || jobban_isbanned(M, "Captain") || jobban_isbanned(M, "Cyborg"))
```
Comment (Birdtalon) [inline code/game/response_team.dm]: More of a policy thing but shouldn't being silicon job-banned just prevent you from being the silicon role? Seems a bit heavy handed to stop them being ERT entirely. Especially if the admin doesn't choose a cyborg anyway.

**Comment by Birdtalon** [comment]
Looks good and should add a bit of variety into ERT spawns which is nice.

**Comment by Kyep** [comment]
@Necaladun Yes. If you bring up the panel and hit 'dispatch' it will simply send a normal RED ert. You don't have to customize anything. The options are simply there if you want to use them.

**Comment by The-Driker** [comment]
Does this allow ghosts to see what positions are being sought when used? If so, I may actually sign up sometimes instead of keeping ERT off.

File: code/game/response_team.dm (hunk 4 above)
Comment (Kyep) [reply code/game/response_team.dm]: ERT is an all-access role with considerable firepower that operates outside the normal chain of command. I got a request to make an Eng job ban block Eng ERT, a sec job ban block sec ERT, etc... but that was too complicated. Also, I am of the opinion that ANY SORT of job ban should prevent you from playing ERT. So, this is a stopgap measure to at least ensure the most problematic types of bans are screened out.

**Comment by Kyep** [comment]
@The-Driker previously no, but I just added that feature based on your suggestion. Ghosts now get a chat notification explaining exactly which slots are open. See second screenshot.

**Comment by The-Driker** [comment]
Excellent. I feel this system will be significantly more user-friendly to everyone involved.

File: code/game/response_team.dm (hunk 4 above)
Comment (Birdtalon) [reply code/game/response_team.dm]: Fair enough 👍

**Comment by taukausanake** [comment]
Man this sounds amazing. Inb4 janitorial ERT is sent for demons to clean up all the blood because the station can't.

Also, not ERP squad. Am disappoint. Would be hilarious as a joke for other things

**Comment by Spartan6** [comment]
@Kyep Is the alert type of the ERT in the ghost notification?

**Comment by Kyep** [comment]
@Spartan6 If you're a ghost when the ERT is created, you see the normal "do you want to join" popup and a line in chat explaining what slots are open (see screenshot).

**Comment by Spartan6** [comment]
Perhaps including the alert level would also be nice. Gamma medics are very different to Red medics and so forth.

**Comment by Kyep** [comment]
@Spartan6 I did not include the ERT level (amber/red/gamma) in the notification that ghosts get, because I (and some other staff) worried that doing so would mean nobody would sign up for Amber ERTs anymore.

**Comment by The-Driker** [comment]
I wouldn't suggest that, you still risk people not signing up if they don't know what they are signing up for. You being worried about people not signing up if it's Amber is more an issue with Amber. 

The issue with Amber being that it's absolutely awful and more often akin to signing up for suicide.

**Comment by msgerbs** [comment]
Do borgs still get a prompt to choose a module? Any way to turn that off for ERT borgs that are going to turn into ERT borgs anyway?

**Comment by Kyep** [comment]
@The-Driker I don't know its fair to dismiss Amber ERT so easily. Amber ERT are stronger than regular sec officers, having better gear and all-access. If people are willing to play regular Sec Officers, they should be willing to play Amber ERT.

@msgerbs All ERT borgs are of the ERT subtype. This subtype means that they are not linked to any AI, require ERT access (not just robotics) to open, hear all radio channels, and don't show up on the robotics console. They also have their own lawset which requires them to obey the ERT commander, but NOT regular crew. Amber and red ERT borgs choose their own module from the normal module list. Gamma ERT borgs don't get to choose their module - they are automatically given the new 'gamma' module, which is like the combat module except it also inherits the traits of ERT borgs. The gamma borg module is very similar to the borgs you get in nations - the ones that look like the combat droids from star wars, having the ability to curl up into a ball and roll around, as well as energy shields.

File: code/game/response_team.dm [hunk 5]
```diff
@@ -129,29 +95,50 @@ var/ert_request_answered = 0
 		if(index > emergencyresponseteamspawn.len)
 			index = 1
 
-		var/client/C = M.client
-		var/mob/living/carbon/human/new_commando = C.create_response_team(emergencyresponseteamspawn[index])
-		new_commando.mind.key = M.key
-		new_commando.key = M.key
-		new_commando.update_icons()
+		spawn(0)
```
Comment (Kyep) [inline code/game/response_team.dm]: Goal: one ERT member being AFK during spawn shouldn't block the rest of the squad from spawning. Would appreciate close review of the logic on this one.

**Pull request merged**